src/main/java/net/sf/jabref/gui/ParserResultWarningDialog.java

```diff
@@ -0,0 +1,90 @@
+/*  Copyright (C) 2003-2015 JabRef contributors.
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.gui;
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.swing.JOptionPane;
+
+import net.sf.jabref.importer.ParserResult;
+import net.sf.jabref.logic.l10n.Localization;
+
+public class ParserResultWarningDialog {
+
+    /**
+     * Shows a dialog with the warnings from an import or open of a file
+     *
+     * @param parserResult - ParserResult for the current import/open
+     * @param jabRefFrame - the JabRefFrame
+     */
+    public static void showParserResultWarningDialog(ParserResult parserResult, JabRefFrame jabRefFrame) {
+        Objects.requireNonNull(parserResult);
+        Objects.requireNonNull(jabRefFrame);
+        showParserResultWarningDialog(parserResult, jabRefFrame, Integer.MAX_VALUE, -1);
+    }
+
+    /**
+     * Shows a dialog with the warnings from an import or open of a file
+     *
+     * @param parserResult - ParserResult for the current import/open
+     * @param jabRefFrame - the JabRefFrame
+     * @param maxWarnings - Maximum number of warnings to display
+     * @param dataBaseNumber - Database tab number to activate when showing the warning dialog
+     */
+    public static void showParserResultWarningDialog(ParserResult parserResult, JabRefFrame jabRefFrame,
+            int maxWarnings, int dataBaseNumber) {
+        Objects.requireNonNull(parserResult);
+        Objects.requireNonNull(jabRefFrame);
+        if (parserResult.hasWarnings()) {
```

Comment:
Semantically, there is no difference. 

But the normal control flow should be obvious to the reader and not hidden in some if branch. The issue with indention is that you always have to remember which condition has lead to this if branch, and you have to have it in mind to understand it. For the normal control flow or the most-used and typical control flow path, it should be obvious and we can put all failures on top, reducing the level of indention and putting the focus on the default control flow.